build.assets/charts/Dockerfile-distroless

```diff
@@ -30,4 +30,7 @@ FROM $BASE_IMAGE
 COPY --from=teleport /opt/staging /
 COPY --from=staging /opt/staging/root /
 COPY --from=staging /opt/staging/status /var/lib/dpkg/status.d
+# Attempt a graceful shutdown by default
+# See https://goteleport.com/docs/reference/signals/ for signal reference.
+STOPSIGNAL SIGQUIT
```

Comment:
There were concerns form @espadolini and a customer that Kubernetes might not honour the stoptignal specified in the image because of https://github.com/kubernetes/kubernetes/issues/30051.

The issue is about supporting a stop signal overload in case the image doesn't properly set the STOPSIGNAL. Kubernetes delegates stopping the pod to the CRI plugin and since 2017 most used CRI implementations (dockershim, containerd, cri-o) support STOPSIGNAL.

I did a test with the following dockerfile
```dockerfile
FROM debian
STOPSIGNAL QUIT
CMD ["bash", "-c", "trap 'echo QUIT signal' QUIT; trap 'echo INT signal' INT; trap 'echo TERM signal' TERM; while true; do echo 'sleeping'; sleep 2; done; exit 0"]
```

and the pod logs do show that we receive the correct signal:
```
sleeping
sleeping
QUIT signal
sleeping
```